docs/safe-haven-services/s3-service.md

```diff
@@ -0,0 +1,214 @@
+```
+export AWS_DEFAULT_REGION=us-east-1
+export AWS_ENDPOINT_URL=http://nsh-fs02:7070
+export AWS_S3_ENDPOINT=http://nsh-fs02:7070
+export AWS_ACCESS_KEY_ID=put_your_key_here
+export AWS_SECRET_ACCESS_KEY=put_your_secret_here
+```
+
+The above could be added to your login script (e.g. `.bashrc`)
+
+Note! If your account has access to R/Python packages via a web proxy then
+you need to temporarily disable it whilst using the S3 service. This is shown
+in the examples below, by setting `http_proxy` to an empty value.
+
+## Use from the command line
+
+First, install the `aws` command:
+```
+pip install awscli
+```
+
+Check this is installed by running `aws`. This should show the command's help documentation.
+If you get `command not found` then run it as `~/.local/bin/aws`
+and consider adding `~/.local/bin` to your `PATH` environment variable.
+
+Make sure your proxy configuration is not applied to the aws command by temporarily disabling it:
+```
+export http_proxy=
+```
+
+Now you can list the contents of a bucket and download a file, for example:
+```
+aws s3  ls  s3://bucket_name
+aws s3  cp  s3://bucket_name/filename.dcm  copy_of_filename.dcm
+```
+
+At this point it will probably complain that it can't locate your credentials.
+There are two ways to supply the information: in environment variables,
+or in a configuration file.
+
+Environment variables can be set using the commmands in the previous section.
+
+Credentials can be stored in a configuration file by running
+```
+aws configure
+```
+and providing the access key, secret key and region (ignore the output format).
+You should also set the endpoint URL, e.g.
+```
+aws configure set --endpoint-url http://nsh-fs02:7070
+```
+
+## Performance tips
+
+* consume the file directly in memory if possible. Saving to disk is not recommended; it wastes disk space and will take 3 times longer to do your processing. See the example code below.
+* If you need to save into a file temporarily (e.g. whilst converting to NIFTI) then save into a RAM disk in `/run/user/$(id -u)/` but delete it straight after use to recover then memory.
+* If it's too large for RAM then save into a file on the system disk, not in your home directory, in `/tmp/$(id)/` but check the disk has space first (using `df -h /tmp/`) and delete it straight after use to recover the disk space.
+
+## Python
+
+### Setup
+
+First, install the `boto3` package:
+```console
+python3 -m virtualenv venv
+source venv/bin/activate
+pip install boto3
+```
+
+Ensure your environment variables are set, as described above.
+The examples below also set `http_proxy` explicitly.
+
+### Download an object
+
+```py
+import boto3
+import os
+os.environ["AWS_ENDPOINT_URL"] = "http://nsh-fs02:7070"
+os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+os.environ["AWS_ACCESS_KEY_ID"] = "dummydata"
+os.environ["AWS_SECRET_ACCESS_KEY"] = "put_your_secret_here"
+os.environ["http_proxy"] = ""
```

Comment:
The user was already told to set the environment variables. Does boto automatically pick these up or is something like [this ](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#passing-credentials-as-parameters) needed?